lib/rbs/prototype/rb.rb

```diff
@@ -1,6 +1,22 @@
 module RBS
   module Prototype
     class RB
+      Context = Struct.new(:module_function, :singleton, keyword_init: true) do
```

Comment:
I introduced `context `for this feature to `prototype rb`, instead of `singleton:` keyword args passing.

Currently it has information about module_function and singleton. And I have a plan to use it for `extend self` because `extend self` needs what is `self` in the context.